### PR TITLE
CHET-183: Update dogfooding on each merge to master

### PR DESCRIPTION
**Pull request by nanux**
I've cleaned up the history of this branch, it was a lot of mess with Github Actions trial and error.

* We need to upload an artifact to share it across jobs
* It needs to be uploaded only from one step from the strategy matrix. I've picked unit tests on Java 11
* The variables are managed from Github secrets
* So far we are doing just a basic check whether the plugin was installed, more extensive tests are coming.
* Branch is now configurable via DOGFOODING_REF github secrets (`refs/heads/master` by default)

### CODE REVIEW

**Comment by bpar476** [comment]
Looks great! is the 'dogfooding ref' part of the configurable branch strategy?

**Comment by nanux** [comment]
> Looks great! is the 'dogfooding ref' part of the configurable branch strategy?

Not sure what you mean by configurable branch strategy. Branch is now configurable via `DOGFOODING_REF` github secrets variable (refs/heads/master by default)

File: .github/workflows/mvn_test.yml [hunk 1]
```diff
@@ -43,3 +46,39 @@ jobs:
           java-version: ${{ matrix.java }}
       - name: Test with maven
         run: mvn -B clean verify -P ${{ matrix.profile}}
+      - name: Upload
+        if: github.ref == env.DOGFOODING_REF && matrix.profile == '!integration' && matrix.java == 11
```
Comment (theghostwhoforks) [inline .github/workflows/mvn_test.yml:50]: Is this to ensure that the artifact is just loaded once?

File: .github/workflows/mvn_test.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ name: Tests
 
 on: [push]
 
+env:
+  DOGFOODING_REF: ${{ secrets.DOGFOODING_REF }}
```
Comment (theghostwhoforks) [inline .github/workflows/mvn_test.yml:6]: is this usually `master` correct? Why is it a secret?

**Comment by theghostwhoforks** [comment]
ah - already merged 👍

File: .github/workflows/mvn_test.yml (hunk 2 above)
Comment (nanux) [reply .github/workflows/mvn_test.yml:6]: It is useful for testing - for example, this branch before merging but potentially in future we can test integration branch or stable tag without changing the code.

**Pull request merged**